vyked/decorators/tcp.py

```diff
@@ -68,6 +68,28 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+def _trace_id_getter(task):
+    trace_id = getattr(task, 'trace_id', None)
+    if trace_id:
+        return trace_id
+    else:
+        try:
```

Comment:
The primary aim is to obtain the trace_id attached to the @api task which has made the request. To do so we need to find the future waiting for the current task to finish, which is done by recursively tracing back (recursive because @api may call a async function which makes call to another function and so on till the final call to request is made).

The numerous try/except are present because the callback function may be a normal function, a partial (hence args[0]) or a method of a task (hence .__self__).